tests/integration/test_pip.py

```diff
@@ -89,3 +91,83 @@ def test_pip_packages(
     _ = utils.fetch_deps_and_check_output(
         tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
     )
+
+
+@pytest.mark.parametrize(
+    "test_params,check_cmd,expected_cmd_output",
+    [
+        # Test case checks fetching pip dependencies, generating environment vars file,
+        # building image with all prepared prerequisites and testing if pip packages are present
+        # in built image
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/pip-e2e-test.git",
+                ref="03939ea78542c7cbfe7a86c5a4133612813e9ec6",
+                packages=(
+                    {
+                        "type": "pip",
+                        "requirements_files": ["requirements.txt"],
+                        "requirements_build_files": ["requirements-build.txt"],
+                    },
+                ),
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            ["python3", "/opt/test_package_cachi2"],
+            "registry.fedoraproject.org/fedora-minimal:36",
+            id="pip_e2e_test",
+        ),
+    ],
+)
+def test_e2e_pip(
+    test_params: utils.TestParameters,
+    check_cmd: List[str],
+    expected_cmd_output: str,
+    cachi2_image: utils.ContainerImage,
+    tmpdir: Path,
+    test_data_dir: Path,
+    request: pytest.FixtureRequest,
+):
+    """
+    End to end test for gomod.
+
+    :param test_params: Test case arguments
+    :param tmpdir: Temp directory for pytest
+    """
+    test_case = request.node.callspec.id
+
+    source_folder = utils.clone_repository(
+        test_params.repo, test_params.ref, f"{test_case}-source", tmpdir
+    )
+
+    output_folder = utils.fetch_deps_and_check_output(
+        tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
+    )
+
+    log.info("Create cachi2.env file")
+    env_vars_file = os.path.join(tmpdir, "cachi2.env")
+    cmd = [
+        "generate-env",
+        output_folder,
+        "--output",
+        env_vars_file,
+        "--for-output-dir",
+        os.path.join("/tmp", f"{test_case}-output"),
+    ]
+    (output, exit_code) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+    assert exit_code == 0, f"Env var file creation failed. output-cmd: {output}"
+
+    log.info("Inject project files")
+    cmd = [
+        "inject-files",
+        output_folder,
+        "--for-output-dir",
+        os.path.join("/tmp", f"{test_case}-output"),
+    ]
+    (output, exit_code) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+    assert exit_code == 0, f"Injecting project files failed. output-cmd: {output}"
```

Comment:
Could this part be in the shared util method as well? This should always be done as part of the build process

Reply:
+1, moved to shared method

Reply:
IMO we don't need the `if package_type == "pip"` condition. The users should always inject-files, regardless of the package manager (otherwise they would have to know which package managers may need that and which ones don't)

Reply:
ou even better, fixed!